pkg/dockerregistry/server/auth.go

```diff
@@ -262,6 +270,9 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 			}
 
 		case "signature":
+			if len(bearerToken) == 0 {
+				return nil, ac.wrapErr(ctx, ErrTokenRequired)
+			}
```

Comment:
these are the cases i was talking about this morning, why do you not also allow these operations to be attempted when no token is provided?

if we're going to do this, the implementation should be such that for *all* api invocations, no token is required and the call succeeds if anonymous permissions are sufficient.